src/kbmod/filters/base_filter.py

```diff
@@ -0,0 +1,18 @@
+import abc
+
+from kbmod.result_list import *
+
+
+class Filter(abc.ABC):
+    """The base class for derived filters on the ResultList."""
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def get_filter_name(self):
+        pass
+
+    @abc.abstractmethod
+    def keep_row(self, row: ResultRow):
+        pass
```

Comment:
These should be documented and targeted to the developer - they need to know what the expected returns are to successfully overwrite the `Filter`. If there are any particular errors in filtering we are willing to catch and ignore (or log) that would make sense to document too. 

We should mention if they are or are not allowed to raise errors and need to handle their own errors as fail-or-pass of the filter. For example `FilterLogLikelihood`, and somehow the likelihood is negative, would raise an `RuntimeError`. This is obviously easily marked as a fail and indicates that something's going wrong with likelihood calculations. While we don't have any logs at the moment, later we need to extend the class to pass it a one - so that the user can at least log the error somewhere. This would be a good place to add that note in the future.